c2corg_api/models/user.py

```diff
@@ -86,14 +94,18 @@ def update_validation_nonce(self, purpose, days):
         particular actions like changing password or validating an email. It
         must have a short lifespan to avoid unused nonces to become a security
         risk."""
+        if purpose != Purpose.registration and not self.email_validated:
+            # An account must be validated before any other action is tried.
+            raise Exception('Account not validated yet')
```

Comment:
No we send a 500 error in this case, which I think is fine for such a special and rare case.